src/router_core/modules/mobile_sync/mobile.c

```diff
@@ -206,6 +206,15 @@ static void qcm_mobile_sync_compose_diff_hint_list(qdrm_mobile_sync_t *msync, qd
 
 static qd_message_t *qcm_mobile_sync_compose_differential_mau(qdrm_mobile_sync_t *msync, const char *address)
 {
+    //
+    // DISPATCH-1738: If the router is shutting down, the qd_dispatch_free() in dispatch.c frees the qd->router_id and
+    // sets the core->router_id to zero. When the core is shutting down, the core->router_id will be zero
+    // and we don't want to proceed with this function. We return here in order to prevent the use after free error on
+    // msync->core->router_id down below in the function.
+    //
+    if (! msync->core->router_id)
+        return 0;
+
```

Comment:
Would it not be better to ensure that the core thread is shut down before freeing the router_id?  If the core thread is running while basic state is being freed, there must be many other potential issues in the code.

Reply:
Can you blank the ID and area at the end of qd_dispatch_free rather than the beginning?

Reply:
ok fair enough. I moved the code that frees the core before the router_id is freed. But I still want to keep the if (! msync->core->router_id)

Reply:
Why?  That's what I'm trying to get rid of.

Reply:
To future proof it if somebody tries to change the order of shutdown just like I did :)

Reply:
This and the one on line 399 are spurious checks that should not be added.  If our standard is to check everything that we _know_ is correct but might someday get changed, then there are about 400 additional checks that you need to put in as well.